app/components/chat/BaseChat.tsx

```diff
@@ -202,6 +156,44 @@ export const BaseChat = React.forwardRef<HTMLDivElement, BaseChatProps>(
       }
     }, []);
 
+    useEffect(() => {
+      let providerSettings: Record<string, IProviderSetting> | undefined = undefined;
+
+      try {
+        const savedProviderSettings = Cookies.get('providers');
+
+        if (savedProviderSettings) {
+          const parsedProviderSettings = JSON.parse(savedProviderSettings);
+
+          if (typeof parsedProviderSettings === 'object' && parsedProviderSettings !== null) {
+            providerSettings = parsedProviderSettings;
+          }
+        }
+      } catch (error) {
+        console.error('Error loading Provider Settings from cookies:', error);
+
+        // Clear invalid cookie data
+        Cookies.remove('providers');
+      }
+
+      let parsedApiKeys: Record<string, string> | undefined = {};
+
+      try {
+        parsedApiKeys = getApiKeysFromCookies();
+        setApiKeys(parsedApiKeys);
+      } catch (error) {
+        console.error('Error loading API keys from cookies:', error);
+
+        // Clear invalid cookie data
+        Cookies.remove('apiKeys');
+      }
+
+      initializeModelList({ apiKeys: parsedApiKeys, providerSettings }).then((modelList) => {
+        console.log('Model List: ', modelList);
+        setModelList(modelList);
+      });
+    }, [apiKeys]);
+
```

Comment:
wont this create an infinite loop?
the only thing calling this hook is this hook itself. 
 the variable `apiKeys ` is only updated by this hook. and also the trigger for the hook

Reply:
I mention this in description, if creation of apiKeys objects is memoized based on input parameters its same object, so it kinda does not change if nothing changed, so no loop